src/NCalc.Core/Parser/LogicalExpressionParser.cs

```diff
@@ -217,16 +217,19 @@ static LogicalExpressionParser()
 
         var charIsNumber = Literals.Pattern(char.IsNumber);
 
+        var dateSeparator = CultureInfo.CurrentCulture.DateTimeFormat.DateSeparator;
+        var timeSeparator = CultureInfo.CurrentCulture.DateTimeFormat.TimeSeparator;
+
         var dateDefinition = charIsNumber
-            .AndSkip(divided)
+            .AndSkip(Literals.Text(dateSeparator))
             .And(charIsNumber)
-            .AndSkip(divided)
+            .AndSkip(Literals.Text(dateSeparator))
             .And(charIsNumber);
 
         // date => number/number/number
-        var date = dateDefinition.Then<LogicalExpression>(static date =>
+        var date = dateDefinition.Then<LogicalExpression>(date =>
         {
-            if (DateTime.TryParse($"{date.Item1}/{date.Item2}/{date.Item3}", out var result))
+            if (DateTime.TryParse($"{date.Item1}{dateSeparator}{date.Item2}{dateSeparator}{date.Item3}", out var result))
```

Comment:
This is inside a static constructor and should run only once in the application life time. But what if in my WebApp I change the current culture? Can you write a unit test checking what happens if you change your current culture?

Reply:
Unfortunately, it was never supported. We can think about it in the future, it can be useful in some cases

Reply:
@gumbarros, to solve this issue we can create a thread-safe dictionary of parsers for each culture and check during the parsing if there is already created parser for specified locale. Passed in Expression locale can be supported too this way. I'll make this changes at the evening.